### PR TITLE
Move Crux specific classes to presto-crux

### PR DESCRIPTION
**Pull request by jainxrohit**
Moving crux analyzer specific classes to presto-crux module. Also renaming native analyzer to crux analyzer.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by jainxrohit** [comment]
> You should try and use the factory pattern and avoid dependencies like this into presto-main

Thanks Sreeni. Right now, presto main depends on many concepts from analyzer world. We can look for more cleanup in future.

File: presto-main/src/main/java/com/facebook/presto/server/ResourceManagerModule.java [hunk 1]
```diff
@@ -87,9 +85,7 @@ protected void setup(Binder binder)
 
         //TODO: Is it really needed here?
         binder.install(new AnalyzerModule());
-        binder.bind(AnalyzerProvider.class).in(Scopes.SINGLETON);
-        binder.bind(BuiltInQueryPreparer.class).in(Scopes.SINGLETON);
-        binder.bind(NativeQueryPreparer.class).in(Scopes.SINGLETON);
+        binder.install(new CruxAnalyzerModule());
```
Comment (ajaygeorge) [inline presto-main/src/main/java/com/facebook/presto/server/ResourceManagerModule.java:88]: curious, why this is needed for Resource Manager?

Comment (jainxrohit) [reply presto-main/src/main/java/com/facebook/presto/server/ResourceManagerModule.java:88]: It's not needed. Thats why I had a TODO as well. However earlier version of resource group module were relying on some analyzer instance. I wonder why it was there.

I will remove it, let me know if you see any concerns. cc: @swapsmagic

Comment (ajaygeorge) [reply presto-main/src/main/java/com/facebook/presto/server/ResourceManagerModule.java]: let's remove it since it is not needed.

**Pull request closed without merging**